autometa/taxonomy/ncbi.py

```diff
@@ -364,28 +338,81 @@ def parse_merged(self):
             logger.debug('merged loaded')
         return merged
 
+    def get_lineage_names(self, taxid):
+        """
+        Returns the lineage names of the taxid
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+
+        Returns
+        -------
+        set
+            lineage of the taxid
+        """
+        lineage = set()
+        while taxid != 1:
+            if self.rank(taxid) not in NCBI.CANONICAL_RANKS:
+                taxid = self.parent(taxid)
+                continue
+            lineage.add(self.name(taxid))
+            taxid = self.parent(taxid)
+        return lineage
+
     def is_common_ancestor(self, parent_taxid, child_taxid):
-        """Determines whether the provided taxids have a non-root common ancestor
+        """
+        Determines whether the provided taxids have a non-root common ancestor
 
         Parameters
         ----------
         parent_taxid : int
-            Description of parameter `parent_taxid`.
+            identifer for a taxon in the Taxonomy Database by NCBI
         child_taxid : int
-            Description of parameter `child_taxid`.
+            identifer for a taxon in the Taxonomy Database by NCBI
 
         Returns
         -------
         boolean
-            True if taxids share a common ancestor and False otherwise
+            True if taxids share a common ancestor else False
         """
-        ancestor_taxid = child_taxid
-        while ancestor_taxid != 1:
-            if parent_taxid == ancestor_taxid:
-                return True
-            ancestor_taxid = self.parent(ancestor_taxid)
+        self.is_valid_taxid(parent_taxid)
+        self.is_valid_taxid(child_taxid)
+        parent_lineage = self.get_lineage_names(parent_taxid)
+        child_lineage = self.get_lineage_names(child_taxid)
+        common_ancestor = child_lineage.intersection(parent_lineage)
+        if common_ancestor:
+            return True
         return False
 
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given taxid is a positive interger or not
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+
+        Returns
+        -------
+        boolean 
+            True if the 'taxid' is a positive interger else False 
+        """
+        if isinstance(taxid, int):
+            return True
+        elif isinstance(taxid, float):
+            return taxid.is_integer()  # This checks if it is something like 12.0 vs. 12.9
```

Comment:
Since you are doing type checking first, you could put all of the logic for a failing taxid first and this would probably allow you to reduce the number of return statements in your method. Similarly, the logging behavior at the end of the method could be incorporated with the checks at the beginning. Also, both of the statements within `elif` and `else` return `False`. Why not have one return outside of the block if they are returning the same value?